whelk-core/src/main/groovy/whelk/search2/QueryParams.java

```diff
@@ -83,57 +82,79 @@ public QueryParams(Map<String, String[]> apiParameters) throws InvalidQueryExcep
         this.aliased = getAliased(apiParameters);
     }
 
-    public Map<String, String> getNonQueryParamsNoOffset() {
-        Map<String, String> params = getNonQueryParams();
-        params.remove(ApiParams.OFFSET);
-        return params;
+    public Map<String, String> getFullParamsMap() {
+        return getParamsMap(List.of(ApiParams.QUERY,
+                ApiParams.SORT,
+                ApiParams.OFFSET,
+                ApiParams.LIMIT,
+                ApiParams.LENS,
+                ApiParams.SPELL,
+                ApiParams.OBJECT,
+                ApiParams.DEBUG,
+                ApiParams.STATS,
+                JsonLd.Platform.COMPUTED_LABEL));
     }
 
-    public Map<String, String> getNonQueryParams() {
-        var params = getParamsMap();
-        params.remove(ApiParams.QUERY);
-        return params;
+    public Map<String, String> getCustomParamsMap(List<String> apiParams) {
+        return getParamsMap(apiParams);
     }
 
-    private Map<String, String> getParamsMap() {
-        if (paramsMap == null) {
-            Map<String, String> params = new LinkedHashMap<>();
+    private Map<String, String> getParamsMap(List<String> apiParams) {
+        Map<String, String> params = new LinkedHashMap<>();
 
-            if (!q.isEmpty()) {
-                params.put(ApiParams.QUERY, q);
-            }
-            if (offset > 0) {
-                params.put(ApiParams.OFFSET, "" + offset);
-            }
-            params.put(ApiParams.LIMIT, "" + limit);
-            if (object != null) {
-                params.put(ApiParams.OBJECT, object);
-            }
-            if (!predicates.isEmpty()) {
-                params.put(ApiParams.PREDICATES, String.join(",", predicates));
-            }
-            var spellP = spell.asString();
-            if (!spellP.isEmpty()) {
-                params.put(ApiParams.SPELL, spellP);
+        for (String param : apiParams) {
+            switch (param) {
+                case ApiParams.QUERY -> {
+                    if (!q.isEmpty()) {
+                        params.put(ApiParams.QUERY, q);
+                    }
+                }
+                case ApiParams.SORT -> {
+                    var sort = sortBy.asString();
+                    if (!sort.isEmpty()) {
+                        params.put(ApiParams.SORT, sort);
+                    }
+                }
+                case ApiParams.OFFSET -> {
+                    if (offset > 0) {
+                        params.put(ApiParams.OFFSET, "" + offset);
+                    }
+                }
+                case ApiParams.LIMIT -> params.put(ApiParams.LIMIT, "" + limit);
+                case ApiParams.LENS -> {} // TODO
```

Comment:
I may have discovered a bug related to the lens parameter. Will investigate it further. No harm in omitting it for now.